nova/virt/vmwareapi/images.py

```diff
@@ -353,18 +354,35 @@ def fetch_image_stream_optimized(context, instance, session, vm_name,
               {'image_ref': image_ref, 'vm_name': vm_name},
               instance=instance)
 
-    metadata = IMAGE_API.get(context, image_ref)
+    metadata = IMAGE_API.get(context, image_ref, include_locations=True)
```

Comment:
We only need the `include_locations` when we enable `CONF.vmware.allow_pulling_images_from_url`, so I would set the argument to the same.